modules/findUnreachableRoutes.js

```diff
@@ -0,0 +1,42 @@
+import { createMemoryHistory } from 'history'
+import warning from './warning'
+
+const createLocation = createMemoryHistory().createLocation
+
+/**
+ * Checks a Route for the component that SHOULD be rendered against the
+ * component that WILL be rendered. If falsy, this indicates an
+ * unreachable route.
+ */
+function verifyRouteIsReachable(path, route, transitionManager) {
+  if (!path || !route) return
+
+  // strip vars so the path becomes routable
+  const cleanPath = path.replace(':', '').replace('*', 'splat')
+
+  transitionManager.match(createLocation(cleanPath), (_, _1, state) => {
```

Comment:
Interesting story: This block causes many of the other tests to fail their assertions, since this runs whenever a test mounts `Router`. Any thoughts on how to resolve this @timdorr or @mjackson? That's my primary remaining blocker for this PR.